### ISSUE TITLE
Handle @typedef with type on wrong side

### ISSUE THREAD

**Issue by andy-ms** [Microsoft/TypeScript#16946]
**TypeScript Version:**  nightly (2.5.0-dev.20170629)

**Code**

```js
// @ts-check

/** @typedef Foo {number} */

/** @type {Foo} */
const x = "";
```

**Expected behavior:**

Either a syntax error, or correctly set `Foo = number` and have a type error.

**Actual behavior:**

No error.

**Comment by errorx666** [comment]
I'd rather have the syntax error.  If we start accepting the wrong syntax, people will start using the wrong syntax.

**Comment by andy-ms** [comment]
Fixed by #18662
